modules/tax/src/Plugin/Commerce/TaxType/CanadianSalesTax.php

```diff
@@ -0,0 +1,261 @@
+<?php
+
+namespace Drupal\commerce_tax\Plugin\Commerce\TaxType;
+
+use Drupal\commerce_order\Entity\OrderInterface;
+use Drupal\commerce_order\Entity\OrderItemInterface;
+use Drupal\commerce_tax\TaxableType;
+use Drupal\commerce_tax\TaxZone;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\profile\Entity\ProfileInterface;
+
+/**
+ * Provides the Canadian tax type.
+ *
+ * @CommerceTaxType(
+ *   id = "CanadianSalesTax",
+ *   label = "Canadian Sales Tax",
+ * )
+ */
+class CanadianSalesTax extends LocalTaxTypeBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+    $form['rates'] = $this->buildRateSummary();
+    $form['rates']['#markup'] = $this->t('The following Canadian tax rates are provided:');
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function shouldRound() {
+    return TRUE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDisplayLabel() {
+    return $this->t('Canadian Sales Tax');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(OrderInterface $order) {
+    $store = $order->getStore();
+    if ($store->getAddress()->getCountryCode() == 'CA') {
+      return TRUE;
+    }
+    $store_registrations = $store->get('tax_registrations')->getValue();
+    $store_registrations = array_column($store_registrations, 'value');
+    if (array_intersect($store_registrations, array('CA'))) {
+      return TRUE;
+    }
+
+    return FALSE;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getZones() {
+    $zones = [];
+
+    $zones['ca'] = new TaxZone([
+      'id' => 'ca',
+      'label' => $this->t('Canadian Government Sales Tax'),
+      'territories' => [
+        ['country_code' => 'CA', 'administrative_area' => 'AB'],
+        ['country_code' => 'CA', 'administrative_area' => 'BC'],
+        ['country_code' => 'CA', 'administrative_area' => 'MB'],
+        ['country_code' => 'CA', 'administrative_area' => 'NT'],
+        ['country_code' => 'CA', 'administrative_area' => 'NU'],
+        ['country_code' => 'CA', 'administrative_area' => 'QC'],
+        ['country_code' => 'CA', 'administrative_area' => 'SK'],
+        ['country_code' => 'CA', 'administrative_area' => 'YT'],
+      ],
+      'rates' => [
+        [
+          'id' => 'gst',
+          'label' => $this->t('GST'),
+          'amounts' => [
+            ['amount' => '0.05', 'start_date' => '2008-01-01'],
+          ],
+          'default' => TRUE,
+        ],
+      ],
+    ]);
+
+    $zones['bc'] = new TaxZone([
+      'id' => 'bc',
+      'label' => $this->t('British Columbia Provincial Sales Tax'),
```

Comment:
Fixed, although GST was hard to label since it is used for some but not all provinces